driver/driver.go

```diff
@@ -278,83 +267,30 @@ func (d *Driver) checkStorageQuota() (map[string]int, error) {
 		return storageDetails, fmt.Errorf("available storage size may be insufficient for correct work of CSI controller")
 	}
 
-	return nil, err
-}
-
-// Stop stops the plugin.
-func (d *Driver) Stop() {
-	d.readyMu.Lock()
-	d.ready = false
-	d.readyMu.Unlock()
-
-	d.log.Info("server stopped")
-	d.srv.Stop()
-}
-
-func WithEndpoint(endpoint string) func(*driverOptions) {
-	return func(o *driverOptions) {
-		o.endpoint = endpoint
-	}
-}
-
-func WithDriverName(driverName string) func(options *driverOptions) {
-	return func(o *driverOptions) {
-		o.driverName = driverName
-	}
-}
-
-func WithAddress(address string) func(options *driverOptions) {
-	return func(o *driverOptions) {
-		o.address = address
-	}
-}
-
-func WithUsername(username string) func(*driverOptions) {
-	return func(o *driverOptions) {
-		o.username = username
-	}
+	return storageDetails, err
 }
 
-func WithPassword(password string) func(*driverOptions) {
-	return func(o *driverOptions) {
-		o.password = password
+func getNodeHost(ctx context.Context, svc *upsvc.Service, nodeHost string) (*upcloud.Server, error) {
+	servers, err := svc.GetServers(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't fetch servers list")
 	}
-}
-
-func WithControllerOn(state bool) func(*driverOptions) {
-	return func(o *driverOptions) {
-		o.isController = state
+	for _, s := range servers.Servers {
+		if nodeHost == s.Hostname {
+			return &s, nil
+		}
 	}
+	return nil, fmt.Errorf("CSI user account doesn't have node with hostname '%s' defined", nodeHost)
 }
 
-func WithNodeHost(nodeHost string) func(*driverOptions) {
-	return func(o *driverOptions) {
-		o.nodeHost = nodeHost
+func logEntyFromOptions(logger *logrus.Logger, options Options) *logrus.Entry {
```

Comment:
Typo?

Reply:
@ajmyyra thanks for review! Good catch, I fixed typos that you found :+1: